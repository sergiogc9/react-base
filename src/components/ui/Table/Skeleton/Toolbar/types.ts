export type TableSkeletonToolbarProps = Record<string, unknown>;
