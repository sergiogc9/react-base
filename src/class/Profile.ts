export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
