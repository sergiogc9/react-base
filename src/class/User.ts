export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
