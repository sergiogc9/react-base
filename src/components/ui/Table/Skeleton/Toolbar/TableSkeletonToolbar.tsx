import React from 'react';
import { Box, Skeleton } from '@sergiogc9/react-ui';

import { TableSkeletonToolbarProps } from './types';

const TableSkeletonToolbar: React.FC<TableSkeletonToolbarProps> = () => {
	return (
		<Box justifyContent="space-between" marginY={3}>
			<Skeleton.Rect height={24} width={150} />
			<Skeleton.Rect height={24} width={200} />
		</Box>
	);
};

export default React.memo(TableSkeletonToolbar);
