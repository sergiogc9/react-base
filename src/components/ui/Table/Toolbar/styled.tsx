import styled from 'styled-components';
import { Box } from '@sergiogc9/react-ui';

import { TableToolbarProps } from './types';

const TableToolbar: React.FC<TableToolbarProps> = styled(Box)<TableToolbarProps>``;

TableToolbar.defaultProps = {
	alignItems: 'center',
	minHeight: 50,
	width: '100%'
};

export default TableToolbar;
