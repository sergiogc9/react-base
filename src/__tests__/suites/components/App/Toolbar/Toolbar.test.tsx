import { shallow } from 'enzyme';
import React from 'react';

import { TenantObject } from '@src/class/Tenant';
import { UserObject } from '@src/class/User';
import Toolbar from '@src/components/App/Toolbar/Toolbar';
import TestHelper from "@src/__tests__/utils/Helper/TestHelper";

const tenant: TenantObject = {
	id: 'rd-girona-test',
	guid: '00034972-0000-0000-0000-000000000000',
	name: 'rd.girona.test',
	tier_properties: {
		name: 'custom',
		results: {
			online: true,
			social: true
		}
	},
	print_only: false,
	facebook_linked_ids: [
		'157193954975917'
	],
	settings: {
		categorization_mode: 'no_flc',
		currency: 'USD',
		display_influencers: true,
		facebook_url: 'https://www.facebook.com/conjunt.chapo',
		valuation_metric: 'miv'
	}
};

const user = TestHelper.getUser();

const userWithoutPermissions: UserObject = {
	...user,
	permissions: []
};

describe('<Toolbar />', () => {
	let wrapper: any;

	beforeEach(() => {
		wrapper = shallow((<Toolbar
			user={user}
			tenant={tenant}
		></Toolbar>));
		jest.useFakeTimers();
	});

	it("Test Toolbar component snapshot with permissions", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test Toolbar component snapshot without permissions", () => {
		wrapper.setProps({ user: userWithoutPermissions });
		expect(wrapper.html()).toMatchSnapshot();
	});
});
