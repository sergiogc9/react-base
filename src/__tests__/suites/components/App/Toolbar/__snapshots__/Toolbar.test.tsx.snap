// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> Test Toolbar component snapshot with permissions 1`] = `"<div id=\\"appToolbar\\" class=\\"md-grid\\"><div class=\\"content md-grid\\"><div class=\\"logo md-cell--middle\\"><img src=\\"/logo_discover.png\\" alt=\\"\\"/><img src=\\"/icon_discover.png\\" class=\\"icon\\" alt=\\"\\"/></div><div class=\\"md-cell--right md-cell--middle\\"><div class=\\"toolbar-buttons md-grid\\"><div id=\\"tenantMenuButton\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below\\"><button id=\\"tenantMenuButton-toggle\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--green-ink\\"><div class=\\"md-ink-container\\"></div>rd.girona.test</button></div><div id=\\"userMenuButton\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below\\"><button id=\\"userMenuButton-toggle\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--green-ink\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons\\">person</i><span>Sergi Massaneda</span></button></div></div></div></div></div>"`;

exports[`<Toolbar /> Test Toolbar component snapshot without permissions 1`] = `"<div id=\\"appToolbar\\" class=\\"md-grid\\"><div class=\\"content md-grid\\"><div class=\\"logo md-cell--middle\\"><img src=\\"/logo_discover.png\\" alt=\\"\\"/><img src=\\"/icon_discover.png\\" class=\\"icon\\" alt=\\"\\"/></div><div class=\\"md-cell--right md-cell--middle\\"><div class=\\"toolbar-buttons md-grid\\"><span>rd.girona.test</span><div id=\\"userMenuButton\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"md-layover md-layover--simplified md-inline-block md-menu-container md-menu-container--menu-below\\"><button id=\\"userMenuButton-toggle\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background btn--green-ink\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons\\">person</i><span>Sergi Massaneda</span></button></div></div></div></div></div>"`;
